spec/models/court_data_spec.rb

```diff
@@ -0,0 +1,304 @@
+require 'rails_helper'
+
+RSpec.describe CourtData do
+  subject(:court_data) { described_class.new(claim_id: claim.id) }
+
+  let(:claim) do
+    create(:claim, case_number: 'TEST12345', create_defendant_and_rep_order: false, defendants: claim_defendants)
+  end
+  let(:claim_defendants) { [] }
+
+  before do
+    search_data.each do |search|
+      allow(LAA::Cda::ProsecutionCase).to receive(:search).with(**search[:params]).and_return(search[:response])
+    end
+  end
+
+  describe '#case_number' do
+    subject { court_data.case_number }
+
+    context 'with a matching case number in Common Platform' do
+      let(:search_data) do
+        [
+          {
+            params: { prosecution_case_reference: 'TEST12345' },
+            response: [instance_double(LAA::Cda::ProsecutionCase, case_number: 'TEST12345')]
+          }
+        ]
+      end
+
+      it { is_expected.to match(hash_including(hmcts: 'TEST12345', claim: 'TEST12345')) }
+    end
+
+    context 'with there is no case number in Common Platform and matching defendant' do
+      let(:search_data) do
+        [
+          {
+            params: { prosecution_case_reference: 'TEST12345' },
+            response: []
+          },
+          {
+            params: { name: 'Zaphod Beeblebrox', date_of_birth: Date.parse('2000-01-01') },
+            response: []
+          },
+          {
+            params: { name: 'Arthur Raffles', date_of_birth: Date.parse('1939-08-17') },
+            response: [hmcts_case]
+          }
+        ]
+      end
+      let(:claim_defendants) do
+        [
+          create(:defendant, first_name: 'Zaphod', last_name: 'Beeblebrox', date_of_birth: Date.parse('2000-01-01')),
+          create(:defendant, first_name: 'Arthur', last_name: 'Raffles', date_of_birth: Date.parse('1939-08-17'))
+        ]
+      end
+      let(:hmcts_case) do
+        instance_double(
+          LAA::Cda::ProsecutionCase,
+          case_number: 'TEST12345',
+          defendants: claim_defendants.map do |defendant|
+            instance_double(
+              LAA::Cda::Defendant,
+              representation_order: instance_double(LAA::Cda::RepresentationOrder,
+                                                    reference: defendant.earliest_representation_order.maat_reference)
+            )
+          end
+        )
+      end
+
+      it { is_expected.to match(hash_including(hmcts: 'TEST12345', claim: 'TEST12345')) }
+    end
+  end
+
+  describe '#status' do
+    subject { court_data.status }
+
+    context 'with a matching case number in Common Platform' do
+      let(:search_data) do
+        [
+          {
+            params: { prosecution_case_reference: 'TEST12345' },
+            response: [instance_double(LAA::Cda::ProsecutionCase, status: 'INACTIVE')]
+          }
+        ]
+      end
+
+      it { is_expected.to match(hash_including(hmcts: 'INACTIVE')) }
+    end
+
+    context 'with there is no case number in Common Platform' do
```

Comment:
Maybe `when there is no case number in Common Platform` or `with no case number in Common Platform`?